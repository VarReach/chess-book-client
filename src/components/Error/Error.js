import React, { Component } from 'react';

export default class Error extends Component {
  render() {
    return (
      <div className="error">
        <span>{this.props.error}</span>
        <button onClick={this.props.hideError}><i className="fas fa-times"/></button>
      </div>
    );
  }
}