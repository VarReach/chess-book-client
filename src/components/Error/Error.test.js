import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Error from './Error';

describe('LoginForm component', () => {
  const props = {
    error: 'Error message',
  };

  it('renders the form by default', () => {
    const wrapper = shallow(<Error/>)
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders the error div given props', () => {
    const wrapper = shallow(<Error {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});