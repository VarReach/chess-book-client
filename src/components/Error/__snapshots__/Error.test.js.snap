// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders the error div given props 1`] = `
<div
  className="error"
>
  <span>
    Error message
  </span>
  <button>
    <i
      className="fas fa-times"
    />
  </button>
</div>
`;

exports[`LoginForm component renders the form by default 1`] = `
<div
  className="error"
>
  <span />
  <button>
    <i
      className="fas fa-times"
    />
  </button>
</div>
`;
